README.rst

```diff
@@ -255,9 +255,114 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
-  annual/monthly_rain/2010.monthly_rain.nc 14
-  annual/monthly_rain/2011.monthly_rain.nc 14
-  annual/monthly_rain/2012.monthly_rain.nc 14
+  annual/monthly_rain/2010.monthly_rain.nc 13
+  annual/monthly_rain/2011.monthly_rain.nc 13
+  annual/monthly_rain/2012.monthly_rain.nc 13
+
+
+Migrating to the New `open` Function
+------------------------------------
+
+Since 1.8.1, there is a `smart_open.open` function that replaces `smart_open.smart_open`.
+The new function offers several advantages over the old one:
+
+- 100% compatible with the built-in `open` function (aka `io.open`): it accepts all
+  the parameters that the built-in `open` accepts.
+- The default open mode is now "r", the same as for the built-in `open`.
+  The default for the old `smart_open.smart_open` function used to be "rb".
+- Fully documented keyword parameters (try `help("smart_open.open")`)
+
+The instructions below will help you migrate to the new function painlessly.
+
+First, update your imports:
+
+.. code-block:: python
+
+  >>> from smart_open import smart_open  # before
+  >>> from smart_open import open  # after
+
+In general, `smart_open` uses `io.open` directly, where possible, so if your
+code already uses `open` for local file I/O, then it will continue to work.
+If you want to continue using the built-in `open` function for e.g. debugging,
+then you can `import smart_open` and use `smart_open.open`.
+
+**The default read mode is now "r" (read text) by default.**
+If your code was implicitly relying on the default mode being "rb" (read
+binary), then you'll need to update it and pass "r" explicitly.
+
+Before:
+
+.. code-block:: python
+
+  >>> buf = b''
+  >>> buf += smart_open('s3://commoncrawl/robots.txt').read(32)
+
+After:
+
+.. code-block:: python
+
+  >>> buf = b''
+  >>> buf += open('s3://commoncrawl/robots.txt', 'rb').read(32)
+
+The `ignore_extension` keyword parameter is now called `ignore_ext`.
+It behaves identically otherwise.
+
+The most significant change is in the handling on keyword parameters for the
+transport layer, e.g. HTTP, S3, etc.  The old function accepted these directly:
+
+.. code-block:: python
+
+  >>> url = 's3://smart-open-py37-benchmark-results/test.txt'
+  >>> session = boto3.Session(profile_name='smart_open')
+  >>> smart_open(url, 'r', session=session).read(32)
+  'first line\nsecond line\nthird lin'
+
+The new function accepts a `transport_params` keyword argument.  It's a dict.
+Put your transport parameters in that dictionary.
+
+.. code-block:: python
+
+  >>> url = 's3://smart-open-py37-benchmark-results/test.txt'
+  >>> session = boto3.Session(profile_name='smart_open')
+  >>> params = {'session': session}
+  >>> open(url, 'r', transport_params=params).read(32)
+  'first line\nsecond line\nthird lin'
+
+Renamed parameters:
+
+- `s3_upload` ➡  `multipart_upload_kwargs`
+- `s3_session` ➡ `session`
+
+Removed parameters:
+
+- `profile_name`
+
+**The `profile_name` parameter has been removed.**
```

Comment:
Something is wrong with the backticks. They have no effect in this section (look like normal text when rendered), and this line actually renders them as literal backticks:

<img width="890" alt="Screen Shot 2019-05-31 at 09 58 03" src="https://user-images.githubusercontent.com/610412/58690768-9c266c00-838a-11e9-9709-376e7ab72532.png">

Reply:
I think that's just the way github renders RST. I don't know what we can do about that.

Reply:
Aaah, this is RST, not MD! That explains it.

Literal blocks in RST use double-backticks IIRC, not single-backtick.

Do we need to use RST though? MD easier? Or is it something to do with PyPI relying on RST…

Reply:
Yes, it's because of PyPI